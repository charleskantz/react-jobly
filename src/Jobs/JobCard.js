import React, { useState } from "react";
import './JobCard.css';
import JoblyApi from "../api/JoblyApi";

// todo: okay to import context here
// todo: use API call to update userInfo instead of pushing new job to userInfo

// details on a specific job, and button to apply for job
function JobCard({
  id,
  title,
  salary,
  equity,
  company_handle,
  username,
  hasApplied,
  setUserInfo
}){
  // state for button
  const[haveApplied, setHaveApplied] = useState(hasApplied);

  // apply for job, update user info with new job application
  const handleApply = evt => {
    async function applyForJob() {
      try {
        let result = await JoblyApi.apply(id, username); // send application to DB
        setHaveApplied(true); // set job as "applied" for button state
        setUserInfo(user => { // push new application to userInfo
          user.jobs.push({
            id,
            title,
            company_handle,
            state: "applied"
          });

          return user;
        })
      }
      catch (err) {
        alert("SOMETHING WENT WRONG!");
        // console.log(err);
      }
    }
    applyForJob();
  }

  return (
    <div className="JobCard" id={id}>
      <h3>{title}</h3>
      <p>Salary: {salary}</p>
      <p>Equity: {equity}</p>
      {haveApplied
        ? <button className="alreadyApplied" disabled>Applied!</button>
        : <button className="buttonApply" onClick={handleApply}>Apply</button>
      }

    </div>
  )
}

export default JobCard