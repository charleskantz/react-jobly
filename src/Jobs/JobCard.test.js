import React from "react";
import { render } from "@testing-library/react";
import JobCard from "./JobCard";

const jobMock = {
  id: 1,
  title: "tester",
  salary: 100,
  equity: 100,
  state: null,
  company_name: "test inc"
}

// smoke
it("renders without crashing", function() {
  render(
    <JobCard job={jobMock} />
  );
});

// snapshot
it("matches snapshot", function() {
  const { asFragment } = render(<JobCard job={jobMock} />);
  expect(asFragment()).toMatchSnapshot();
});