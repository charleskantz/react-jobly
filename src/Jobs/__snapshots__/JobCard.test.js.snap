// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1pu80g8"
  >
    <div
      class="css-1juiejp"
    >
      <h1
        class="css-1awjs8x"
      >
        tester
      </h1>
      <p
        class="css-zsarox"
      >
        test inc
      </p>
      <p
        class="css-zsarox"
      >
        Salary: $100
      </p>
      <p
        class="css-zsarox"
      >
        Equity: 10000%
      </p>
    </div>
    <button
      class="buttonApply css-1xrqucq"
    >
      Apply
    </button>
  </div>
</DocumentFragment>
`;
